### ISSUE TITLE
On-the-Fly Image Manipulations Don't Update When An Image Is Replaced

### ISSUE THREAD

**Issue by pyrobob83** [ExpressionEngine/ExpressionEngine#740]
TL;DR :- On-the-Fly Image Manipulations Don't Update When An Image Is Replaced

The 'cached' image manipulations in the /_resize/ directory that are created on-the-fly via the template (eg `<img src="{content:file:resize width='100' height='100'}" alt="" />`) do not clear/update if a CMS user **replaces** the original file via the standard image conflict resolution dialogue.

When using traditional directory-defined manipulations, each of the setup manipulations is re-generated at the same time the original image is **_replaced_**, but this does not happen with the ones generated on-the-fly - the initial cached/resized image is still output in the HTML.

Since they're generated on-the-fly, presumably the solution is to just delete any auto-generated versions when an original is updated and the new generated version will just get re-created on next request.

I'm also wondering how tag caching comes into play here... with the directory-based manipulations, the URL of specifically sized images doesn't change, so if an `{exp:channel:entries}` tag has caching on, there's no issue if an image is _**replaced**_ since the manipulations also get replaced without change to the manipulation URLs.  The on-the-fly /_resize/ versions have unique  and seemingly-random identifiers appended to the filenames, so deleting a /_resize/ manipulation may not work.

I'll let you guys figure out the best way to do it, but ultimately I think the versions created on the fly also need to update if an image is replaced (on next request is fine IMHO).

**Comment by intoeetive** [comment]
@pyrobob83 thanks for the detailed report - we'll get this fixed in 6.0.1

Thank you for deeper thinking on template tag caching as well! This is not an issue however - the seemingly-random part of file name is not totally random but represents the combination of manipulation parameters as defined in template, so it will not change with file update

**Comment by ops-andy** [comment]
fixed in 6.0.1
